### DISCUSSION TITLE
assigning a channel to a facet

### DISCUSSION THREAD

**Discussion by tlodge** [vendurehq/vendure#1540] [Q&A] [answered]
Hi, I'm creating a multivendor store and have managed to get most of the way there.  However my usecase requires that all vendors  share the same shopfront, and can create new products under a shared facet structure.  

I have a plugin with an eventbus subscription that listens for  newly created admins and then creates a new channel.  It then attempts to add that channel to each of the site's facets.  I could do this by updating the underlying database directly, but would rather do it through using vendure's various services.  What I currently have is this:

```
 const input = {
  code,
  token: gentoken(18),
  defaultLanguageCode: LanguageCode.en,
  pricesIncludeTax: true,
  currencyCode: CurrencyCode.GBP,
  defaultTaxZoneId: 2,
  defaultShippingZoneId: 2,
}

const channel = await this.channelService.create(ctx, input) 

if (isGraphQlErrorResult(channel)) {
  return; //handle this later..
}

const facets = await this.facetService.findAll(ctx);

for (const facet of facets.items){
     const finput = facet;  //facet that we want to update with the new channel              
     finput.channels = [...finput.channels, channel];  //<--- typescript complains about this attempt to append the new channel
     this.facetService.update(ctx, finput);
} 

```

But typescript complains because it is expecting the new channel that I want to add to be an intersection type between Channel and a bunch of translation stuff, i.e. I get the following:

```
Type 'Channel' is not assignable to type 'Channel & { languageCode: LanguageCode; } & { id: Translated<ID>; createdAt: Translated<Date>; updatedAt: Translated<Date>; defaultTaxZone: Translated<...>; defaultShippingZone: Translated<...>; }'.ts(2322)
```    

I'm not too sure why it's expecting this, or if I need to create this new type (of if I shouldn't be doing this..).

Any thoughts or pointers would be very gratefully received!

Thank you.

**Comment by michaelbromley** [comment, marked as answer]
Hi,

the code looks good, and the reason TS complains is that the `facetService.findAll()` method returns `Translated<Facet>` which recursively affects nested entities like Channel too. In this case it is totally safe to use an assertion `as Channel` to make TS happy.

**Reply by tlodge** [reply]
Thank you!  Really appreciate the quick response.  That's great.
